src/transformers/pipelines/automatic_speech_recognition.py

```diff
@@ -311,14 +311,14 @@ def _sanitize_parameters(
 
         forward_params = defaultdict(dict)
         if max_new_tokens is not None:
-            forward_params["generate_kwargs"]["max_new_tokens"] = max_new_tokens
+            forward_params["max_new_tokens"] = max_new_tokens
```

Comment:
Note in this file's diff, also applicable for the diff in `src/transformers/pipelines/image_to_text.py`: 

The conventional strategy to pass kwargs to `generate` is through `**forward_params`. Previously in this file, the generation kwargs were held as `forward_params["generate_kwargs"]`, which prevented the use of the conventional strategy. There isn't really a reason to hold these kwargs separately, `generate` is the only sink for kwargs in models that can generate. Models that can't generate will throw an exception regardless of the container for kwargs. As such, this diff aims at minimizing the difference across pipelines :)

Reply:
Actually - I realise what I've said about the forward kwargs is wrong here - we can just assume they're passed to the model. In this case, my preference is to still have "generate_kwargs" explicitly in the forward_kwargs, but I don't feel strongly and don't mind if you leave as-is

Reply:
I think we can agree that regardless of the pattern we choose here, it should be applied to all pipelines with generative capabilities for consistency. Based on this premise, enforcing a separation of `generate_kwargs` this exact way will break backward compatibility, i.e. the following would not be possible

```py
from transformers import pipeline

llm = pipeline(task='text-generation', model="openai-community/gpt2")
response = llm('The capital of France ', max_length=50)
```

Nevertheless, I am aligned with you -- we should separate them! We can do it through `generation_config.update(**kwargs)`, and perform the required validation with the aid of `generation_config.validate()`. One of the requirements to do so is to have a single big blob of keyword arguments to untangle, and thus these changes go in this direction.

Let me know if you agree, in which case I'll merge the PR and prepare this follow-up. [My instinct was to merge this PR now, but I've held it back -- I've merged too many not-100%-approved PRs recently 😉 ]

Reply:
Yeah, let's merge atm so this is unblocked and then we can iterate on something different :)